### PR TITLE
Prevent token collection corruption by fixers

### PR DESCRIPTION
**Pull request by keradus**
Continuation of #1151 for 2.x line

### CODE REVIEW

**Comment by keradus** [comment]
cc @stof

File: Symfony/CS/Tests/Fixer/AbstractFixerTestBase.php [hunk 1]
```diff
@@ -69,26 +69,24 @@ protected function makeTest($expected, $input = null, \SplFileInfo $file = null,
...
         $this->assertSame($expected, $tokens->generateCode(), 'Code build on expected code must not change.');
-        $this->assertFalse($tokens->isChanged(), 'Tokens collection built on expected code should not be marked as changed after fixing.');
     }
 
-    private function assertTokens(Tokens $expectedTokens, Tokens $tokens)
+    private function assertTokens(Tokens $expectedTokens, Tokens $inputTokens)
     {
-        foreach ($expectedTokens as $index => $expectedToken) {
-            $token = $tokens[$index];
+        $this->assertEquals($expectedTokens->count(), $inputTokens->count(), 'The collection must have the same length than the expected one.');
```
Comment (stof) [inline Symfony/CS/Tests/Fixer/AbstractFixerTestBase.php]: checking the length at the end was actually expected, to have a better chance to see the difference between collections, to help debugging (if you only know that the length is different, it is hard to debug this)

**Comment by stof** [comment]
@keradus I have some fixes locally. I will send a PR to your fork

File: Symfony/CS/Tests/Fixer/AbstractFixerTestBase.php (hunk 1 above)
Comment (keradus) [inline Symfony/CS/Tests/Fixer/AbstractFixerTestBase.php]: good point, done

**Comment by keradus** [comment]
That would be great!

**Comment by stof** [comment]
PR sent: https://github.com/keradus/PHP-CS-Fixer/pull/10 Tests are back to green

**Comment by keradus** [comment]
Thank you!
Looks ready to merge

**Comment by gharlan** [comment]
:+1:

**Comment by stof** [comment]
Looks good to me (but all fixes are mine, so I'm not the right one to validate the PR)

**Comment by keradus** [comment]
:+1: by me too

**Comment by keradus** [comment]
Thank you @stof.

**Comment by keradus** [comment]
> Assertions: 25233

We are so crazy :D

**Pull request merged**